### PR TITLE
VC-4235: added assets states

### PR DESCRIPTION
**Pull request by vovan1983**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: database/vipcoin/chain/assets/assets.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	}
+}
+
+// SaveAssets - method that save assets to the "vipcoin_chain_assets_assets" table
+func (r Repository) SaveAssets(assets ...*assetstypes.Asset) error {
+	if len(assets) == 0 {
+		return nil
+	}
+
+	tx, err := r.db.BeginTxx(context.Background(), &sql.TxOptions{})
```
Comment (alex-bodnar) [inline database/vipcoin/chain/assets/assets.go]: Можно упростить до:


	query := `INSERT INTO vipcoin_chain_assets_assets 
			(issuer, name, policies, state, issued, burned,
        withdrawn, in_circulation, precision, fee_percent, extras) 
		VALUES 
			(:issuer, :name, :policies, :state, :issued, :burned,
        :withdrawn, :in_circulation, :precision, :fee_percent, :extras)`

	if _, err := r.db.NamedExec(query, toAssetsDatabase(msg...)); err != nil {
		return err
	}

	return nil
}

File: database/vipcoin/chain/assets/assets.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	return assets, nil
+}
+
+// UpdateAssets - method that updates the assets in the "vipcoin_chain_assets_assets" table
+func (r Repository) UpdateAssets(assets ...*assetstypes.Asset) error {
+	if len(assets) == 0 {
+		return nil
+	}
+
+	tx, err := r.db.BeginTxx(context.Background(), &sql.TxOptions{})
```
Comment (alex-bodnar) [inline database/vipcoin/chain/assets/assets.go:79]: Так же можно упростить.

File: database/vipcoin/chain/assets/helpers.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	result := make([]extratypes.Extra, 0, len(extras))
+	for _, extra := range extras {
+		result = append(result, *extra)
+	}
+
+	return types.ExtraDB{Extras: result}
+}
+
+// fromExtrasDB - mapping func from database model
+func fromExtrasDB(extras types.ExtraDB) []*extratypes.Extra {
```
Comment (alex-bodnar) [inline database/vipcoin/chain/assets/helpers.go:22]: Проверь эту функцию, скорее всего при конвертации масива тут будет ошибка из-за указателя.

File: database/types/vipcoin_assets.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+	// DBAssetCreate represents a single row inside the "vipcoin_chain_assets_create" table
+	DBAssetCreate struct {
+		Creator    string        `db:"creator"`
+		Name       string        `db:"name"`
+		Issuer     string        `db:"issuer"`
+		Policies   pq.Int32Array `db:"policies"`
+		State      int32         `db:"state"`
+		Precision  int64         `db:"precision"`
+		FeePercent int64         `db:"fee_percent "`
```
Comment (IamStubborN) [inline database/types/vipcoin_assets.go]:
```suggestion
		FeePercent int64         `db:"fee_percent"`
```

File: database/types/vipcoin_assets.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	}
+
+	// DBAssetManage represents a single row inside the "vipcoin_chain_assets_manage" table
+	DBAssetManage struct {
+		Creator       string        `db:"creator"`
+		Name          string        `db:"name"`
+		Policies      pq.Int32Array `db:"policies"`
+		State         int32         `db:"state"`
+		Precision     int64         `db:"precision"`
+		FeePercent    int64         `db:"fee_percent "`
```
Comment (IamStubborN) [inline database/types/vipcoin_assets.go]:
```suggestion
		FeePercent    int64         `db:"fee_percent"`
```

**Review by alex-bodnar — APPROVED** [review]
LGTM

**Review by IamStubborN — APPROVED** [review]

**Comment by IamStubborN** [comment]
LGTM

**Pull request merged**